### PR TITLE
Add new widget alarmeta

### PR DESCRIPTION
**Pull request by nxdefiant**
This widget displays the time to the next Alarm or Timer in hours and minutes.

### CODE REVIEW

File: apps/alarmeta/metadata.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "id": "alarmeta",
```
Comment (alessandrococco) [inline apps/alarmeta/metadata.json:2]: my 2 cents: this is a widget so the name [_should_ start with **wid**](https://github.com/espruino/BangleApps#example-applications)

File: apps/alarmeta/widget.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    if (next > 0 && next < 86400000) {
+      const hours = Math.floor((next % 86400000) / 3600000).toString();
+      const minutes = Math.floor(((next % 86400000) % 3600000) / 60000).toString();
+
+      g.reset(); // reset the graphics context to defaults (color/font/etc)
+      g.setFontAlign(0,0); // center fonts
+      g.clearRect(this.x, this.y, this.x+this.width-1, this.y+23);
+
+      // Use 'locale' module to get a shortened month name
+      // in the correct language
```
Comment (alessandrococco) [inline apps/alarmeta/widget.js:16]: You could use the new locale_utils, see https://github.com/espruino/BangleApps/pull/1784

Comment (nxdefiant) [reply apps/alarmeta/widget.js:16]: Leftover comment from the template in http://www.espruino.com/Bangle.js+Widgets

File: apps/alarmeta/metadata.json (hunk 1 above)
Comment (nxdefiant) [reply apps/alarmeta/metadata.json]: done

**Comment by gfwilliams** [comment]
Thanks! Just merging, but just a note:

> `const alarms = require("Storage").readJSON("sched.json",1) || [];`

This is going to load all alarms into RAM and keep them there. Not a huge issue for Bangle.js 2, but on Bangle.js 1 maybe it matters more.

You *could* just work out the time to the next alarm at startup, and store that. Then to work around the issue with days of the week you queue another check just after midnight.

**Pull request merged**